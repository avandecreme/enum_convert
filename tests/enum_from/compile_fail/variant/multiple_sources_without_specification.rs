use enum_convert::EnumFrom;

enum First {
    Unit,
}

enum Second {
    Unit,
}

#[derive(EnumFrom)]
#[enum_from(First, Second)]
enum Target {
    #[enum_from] // Ambiguous - should specify which enum when multiple sources
    Unit,
}

fn main() {}
