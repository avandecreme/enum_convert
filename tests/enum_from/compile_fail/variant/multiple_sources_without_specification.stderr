error: When multiple source enums are specified, each variant must specify from which enum to convert with #[enum_from(Enum)] or #[enum_from(Enum::Variant)]
  --> tests/enum_from/compile_fail/variant/multiple_sources_without_specification.rs:14:5
   |
14 |     #[enum_from] // Ambiguous - should specify which enum when multiple sources
   |     ^
